### DISCUSSION TITLE
The Ideal LLM App...

### DISCUSSION THREAD

**Discussion by rb81** [intellibar/main#41] [Ideas]
**** Sharing for others to contribute. ****

Here's what I think would make the ideal LLM app, divided into two sets of features `standard` and `advanced`:

## Standard

- Ability to choose specific providers (e.g., OpenAI, Claude, Gemini, etc.)
- Ability to choose specific models (e.g., OpenAI: gpt-3.5-turbo, gpt-4-turbo, fine-tuned models, etc.)
- Integration with Ollama to use local models
- Return of the Chat window to properly manage conversations
- Ability to switch models during conversation

## Advanced

- Ability to alter hyperparameters and system prompt _per conversation_
- Ability to import/export conversations
- Ability to share conversations (exports entire thread as HTML/CSS or PDF to share with others)
- RAG features with both external (e.g., OpenAI) and Ollama-served embedding models
- Transparency of IntelliBar system prompts (e.g., when using RAG)

**Note:** I'll update this list as things come to mind.

**Comment by mlpro1122** [comment]
GPT-4 is ideal LLM app till now
